contracts/src/v0.8/dev/vrf/VRFV2PlusWrapper.sol

```diff
@@ -384,11 +386,23 @@ contract VRFV2PlusWrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsume
     lastRequestId = requestId;
   }
 
+  function removeDomainSeparator(bytes memory bytesInput) public view returns (bytes memory newBytes) {
+    newBytes = new bytes(bytesInput.length);
+    for (uint8 i = 4; i < 36; ++i) {
+      newBytes[i - 4] = bytesInput[i];
+    }
+    return newBytes;
+  }
+
   function requestRandomWordsInNative(
     uint32 _callbackGasLimit,
     uint16 _requestConfirmations,
-    uint32 _numWords
+    uint32 _numWords,
+    bytes memory extraArgs
   ) external payable override returns (uint256 requestId) {
+    bool nativePayment = abi.decode(removeDomainSeparator(extraArgs), (bool));
+    require(nativePayment, "nativePayment not true in requestRandomWordsInNative call");
```

Comment:
Similar to the above comment:

```suggestion
	if (extraArgs.length < 36) {
		revert IncorrectExtraArgsLength(EXPECTED_LENGTH, extraArgs.length);
	}
    bool nativePayment = extraArgs[35] == 1
    if (nativePayment) {
    	revert NativePaymentInOnTokenTransfer();
    }
```

I think you can probably extract the extraArgs check and the nativePayment decode into a helper and use that in both functions to keep things DRY and you can also test this helper to make sure it works properly. Can also do gas usage tests on the helper directly if needed.

Reply:
Fixed this by adding a common method checkPaymentMode